From ddc159689dd4f10da50e5d91bcf100e8da8f857d Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Wed, 3 Sep 2025 16:17:36 +0200
Subject: [PATCH 004/239] CMake: Work around looking for Qt packages in in-tree
 tests

Calling find_package(Qt6 COMPONENTS Qml) inside a tests/
CMakeLists.txt with an in-tree tests build will fail while trying to
include the Qt6QmlToolsTargets.cmake file, because the file is
generated at generation time, but we try to include at configure time.

This works in 6.11+, because we have code that tries to detect not
having to include the Targets file if not needed.

Work around the issue by setting QT_NO_CREATE_TARGETS to ON before the
find_package call in the qmltc doc snippet.

Problematic code originally introduced in
5296c0255b83b8a92e82b3ec4d536655742b90cf in dev.
Amends cherry-pick dad91592e609364231decaba771418d2d48d4910

Change-Id: I5ff510c58e9c50cc3d9e8cde999a8ab531debdd5
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
---
 src/qml/doc/snippets/qmltc/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/qml/doc/snippets/qmltc/CMakeLists.txt b/src/qml/doc/snippets/qmltc/CMakeLists.txt
index 93477a70cc..15a574c855 100644
--- a/src/qml/doc/snippets/qmltc/CMakeLists.txt
+++ b/src/qml/doc/snippets/qmltc/CMakeLists.txt
@@ -57,6 +57,10 @@ qt6_add_qml_module(${application_name}
 )
 #! [qmltc-add-qml-module]
 
+# Work around not being able to call find_package() with Qt components
+# when building qtdeclarative with in-tree tests.
+set(QT_NO_CREATE_TARGETS ON)
+
 #! [qmltc-compile-to-cpp]
 # (qmltc-specific) Link *private* libraries that correspond to QML modules:
 find_package(Qt6 COMPONENTS QmlPrivate QuickPrivate)
-- 
2.51.2

